### PR TITLE
[Azure Search][Preview] Stabilize state of feature/search/preview branch

### PR DESCRIPTION
**Pull request by arv100kri**
`feature/search/preview` is going to be the (temporary) ship vehicle for an eventual preview SDK for Azure search (tentatively 11.0.0-preview)

This commit does the following:

1. Regenerates code from swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/7615 that introduced a "base swagger"
2. Re-adds features that were in `2017-11-11-Preview` (and will be in `2019-05-06-Preview`) but not in `2019-05-06`. The only notable change is customer managed encrypted indexes/synonym maps

This is *not* the final version of the preview SDK - we intend to add features (via the swagger spec, to keep it in sync with the REST API). We intend to remove some deprecated types as well.

This PR, sets us up to allow that to happen.

(Optionally) after this PR is checked in, the branch `search-preview` can be eliminated.

### CODE REVIEW

File: sdk/search/Microsoft.Azure.Search/tests/SessionRecords/EncryptionWithCustomerManagedKeysTests/CreateIndexWithCustomerManagedKeysUsingAccessCredentials.json [hunk 1]
```diff
@@ -7,48 +7,48 @@
       "RequestBody": "",
       "RequestHeaders": {
         "x-ms-client-request-id": [
-          "d9f26adf-bdc6-4251-a0ee-cf3e57bfdb7d"
+          "458eff35-1b96-411c-adb3-f112e2ef3dfc"
         ],
-        "accept-language": [
```
Comment (bleroy) [inline sdk/search/Microsoft.Azure.Search/tests/SessionRecords/EncryptionWithCustomerManagedKeysTests/CreateIndexWithCustomerManagedKeysUsingAccessCredentials.json]: Any particular reason for the casing to change here?

**Review by bleroy — APPROVED** [review]
LGTM

File: sdk/search/Microsoft.Azure.Search/tests/SessionRecords/EncryptionWithCustomerManagedKeysTests/CreateIndexWithCustomerManagedKeysUsingAccessCredentials.json (hunk 1 above)
Comment (arv100kri) [reply sdk/search/Microsoft.Azure.Search/tests/SessionRecords/EncryptionWithCustomerManagedKeysTests/CreateIndexWithCustomerManagedKeysUsingAccessCredentials.json]: None, but I think it doesn't matter for http headers... This is auto-generated by the test tool, so I didn't hand edit it

**Comment by arv100kri** [comment]
Need to add session records for all tests, not just the tests that I added.
Will do that and push another commit

**Comment by arv100kri** [comment]
> Need to add session records for all tests, not just the tests that I added.
> Will do that and push another commit

Fixed with a recent commit. Will check-in if the tests pass

**Pull request merged**